btfxwss/queue_processor.py

```diff
@@ -120,19 +101,15 @@ def run(self):
             elif dtype == 'data':
                 try:
                     channel_id = data[0]
-                    if channel_id != 0:
-                        # Get channel type associated with this data to the
-                        # associated data type (from 'data' to
-                        # 'book', 'ticker' or similar
-                        channel_type, *_ = self.channel_directory[channel_id]
-
-                        # Run the associated data handler for this channel type.
-                        self._data_handlers[channel_type](channel_type, data, ts)
-                        # Update time stamps.
-                        self.update_timestamps(channel_id, ts)
```

Comment:
This line I did not quite understand why it was inside the if statement.

Reply:
This was a later addition (since the lib initially didnt support the authenticated channels). Since the `auth` channel has various "sub-types" I wanted to sort, I did this to avoid cluttering the run method further (and moved the sorting to `self._handle_account()`. That made the look up in `channel_directory` superfluous, hence the `if`. 

Not very elegant, but most of the code was a kind of on-the-fly implementation, as I needed it. So if that doesn't make much sense to you, that's probably because it doesn't ;D